compat/s3-tests/patches/apply-nuke-fallback.py

```
@@ -0,0 +1,145 @@
+                        UploadId=u['UploadId'],
+                    )
+                except ClientError:
+                    pass
+        except ClientError:
+            pass
+
+        client.delete_bucket(Bucket=bucket)
+    except ClientError as e:
+        # Leftover multiparts or other state: log and continue.
+        # Tests use unique prefixes; a stuck bucket must not halt the suite.
+        print(f"  warning: could not delete bucket {bucket}: {e}")
```

Ruff (0.15.21)

[error] Undefined name `bucket`

(F821)